### ISSUE TITLE
windows.h included by vulkan.h

### ISSUE THREAD

**Issue by redorav** [KhronosGroup/Vulkan-Docs#774]
I have using Vulkan on Windows for a little while now, toying around and learning about it. I have recently refactored a bit of code which has caused another completely unrelated part of the code to not compile. This code uses std::max, which windows.h was defining. The windows header is being included by vulkan.h with no extras. Previously I didn't have an issue because NOMINMAX had been defined elsewhere in the code and it was rippling through. After the refactor the define was no longer there.

Including the define locally in my code fixes the issue, but I wonder if the Vulkan headers themselves shouldn't define things like NOMINMAX, WIN32_LEAN_AND_MEAN, VC_EXTRALEAN, etc. all of the things that it probably doesn't need and that are bloating the header on Windows. What do you think?

**Comment by pdaniell-nv** [comment]
If you want more control over what the Vulkan headers include use vulkan_core.h and vulkan_win32.h instead.

**Comment by oddhack** [comment]
Agreed with @pdaniell-nv - that's why we created the window system-specific headers in the first place. That said, there's a small set of Windows native types/definitions that the Vulkan Windows APIs need, I think just 

* HINSTANCE
* HWND
* HANDLE
* SECURITY_ATTRIBUTES
* DWORD
* LPCWSTR

at present. Possibly all of them could be accommodated with a different (simpler) system header, although there's no way to know what native types might be needed in the future.

**Comment by redorav** [comment]
Thanks a lot for your replies! I tried simply typedef'ing the types to what windows defines and it works fine and doesn't need to pull the windows headers in. I suppose I can do like you say and customize my includes. It's still a bit of a shame that such a lightweight header becomes very big when it doesn't really need to, and a little unexpected. I only realized it was a problem when compilation started failing.
